File: callreco-agent/src/main/java/com/williamfzc/callreco/agent/storage/CrStorage.java
package com.williamfzc.callreco.agent.storage;

import java.util.HashMap;
import java.util.Map;

public class CrStorage {
    public static final CrStorage INSTANCE;

    private final Map<Long, CrProbeUnit> data = new HashMap<>();

    public CrProbeUnit getProbeUnit(long classId, int probesCount) {
        CrProbeUnit unit = data.get(classId);
        if (null != unit) {
            return unit;
        }

        // create a new one
        unit = new CrProbeUnit(classId, probesCount);
        data.put(classId, unit);
        return unit;
    }

    static {
        INSTANCE = new CrStorage();
    }
}

File: callreco-agent/src/main/kotlin/com/williamfzc/callreco/agent/storage/CrStorage.kt
package com.williamfzc.callreco.agent.storage

object CrStorage {
    private val data = mutableMapOf<Long, CrProbeUnit>()

    fun getProbeUnit(classId: Long, probesCount: Int = -1): CrProbeUnit {
        data[classId]?.let { existed ->
            return existed
        }

        CrProbeUnit(classId, probesCount).let { newUnit ->
            data[classId] = newUnit
            return newUnit
        }
    }
}